### PR TITLE
Make sure that the addon agent image is same with hub image and change the addon name to `submariner`

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: liuwei <liuweixa@redhat.com>

### CODE REVIEW

File: pkg/hub/submarineraddonagent/agent.go [hunk 1]
```diff
@@ -187,3 +208,18 @@ func (a *addOnAgent) permissionConfig(cluster *clusterv1.ManagedCluster, addon *
 
 	return operatorhelpers.NewMultiLineAggregate(errs)
 }
+
+func (a *addOnAgent) getCurrentPodImage() (string, error) {
+	podNamespce := os.Getenv("POD_NAMESPACE")
```
Comment (qiujian16) [inline pkg/hub/submarineraddonagent/agent.go:213]: i think we only needs to call this once before starting controller. And i think it is helpful if we have an agent-image flag. If it  is not set, we fall back to use this method to get image. if it fails,  we fatal the whole container.

Comment (skeeey) [reply pkg/hub/submarineraddonagent/agent.go]: done

**Comment by skeeey** [comment]
refer to https://github.com/open-cluster-management/backlog/issues/6440

**Review by qiujian16 — COMMENTED** [review]
/lgtm

**Pull request merged**